### DISCUSSION TITLE
"Level"-Werte der Thermostate zusätzlich auch als (disabled) Sensor für Langzeitstatistik zur Verfügung stellen

### DISCUSSION THREAD

**Discussion by Bascht74** [SukramJ/aiohomematic#695] [Ideas]
**Is your feature request related to a problem? Please describe.**
Akuell werden die Stellungen der Heizthermostate als "Input"-Entität in Home Assistant zur Verfügung gestellt. 
Das ist super, wiel man die Werte dadurch vorgeben kann (auch wenn sie nach kurzer Zeit wieder überschrieben werden). 
Allerdings gibt es für Input-Entitäten keine Langzeitstatistik. Das merkt man schon, wenn man ab HA 2022.11.x in einem Dashboard die Entität öffnet. Oder wenn man eben die neuen Lanzeit-Karten verwenden möchte.

**Describe the solution you'd like**
Es wäre ganz prima, wenn die Level-Entitäten (die ja standardmäßig disabled sind) zusätzlich auch als normale Sensor-Entität zur Verfügung stehen würden (als Klasse "Measurement"). Dadurch wären diese Werte auch in der Langzeitstatistik verfügbar. Die Entitäten könnten standardmäßig ebenfalls disabled sein. 

**Describe alternatives you've considered**
Diese Maßnahme würde es einem ersparen, zusätzliche template Sensor Entitäten anzulegen, die den Zustand übernehmen und so zu konfigurieren, dass sie in der Langzeitstatistik aufgenommen werden.
Ich kann das als "Workaround" jetzt so auch lösen. Ich frage nur nach, ob hier einer der Masterminds der tollen Integration nicht auch inzwischen einen Faible für die Langzeitstatistik entwickelt hat. 

**Additional context**
Vielleicht betrifft das im Kontext auch andere Input-Sensoren, wo das Sinn macht. 
Seit HA 2022.11 hat sich bezgl. der Langzeitstatistik schon etwas getan.
Ich hoffe, dass eine Umsetzung über die Integration nur eine kleine Zauberei darstellt :-)

**Comment by SukramJ** [comment]
Vielen Dank  für deinen _Feature Request._

So wie ich das sehe geht es nur um den Parameter `LEVEL` bei TRVs, der als **Number**-Entität _automatisch generiert_ wird. Um aus den _automatisch generierten_ Entitäten zusätzliche Sensoren zu erzeugen muss daher eine CustomEntity für Sensoren erstellt werden, für die es derzeit keinen weiteren Anwendungsfall gibt.

Für deinen Anwendungswall gibt es, so wie du es bereits verwendest, Template-Entitäten um eigene Anpassungen vorzunehmen. Der einmalige Aufwand pro TRV ein Template zu erstellen scheint mir doch sehr begrenzt.

Ich werde daher keine zusätzliche Implementierung für LEVEL-Sensoren erstellen.

**Comment by Bascht74** [comment]
@SukramJ 
Danke für deine Antwort. Genau es geht um die TRVs, hätte ich schreiben können. 
Dann mache ich das so mit den Template-Variablen, die Anzahl ändert sich ja nicht dauernd :-)

Allerdings habe ich den use case für diese input Variablen nicht verstanden. 
Wenn ich hier einen Wert für den "Level" setze, bleibt der einen bestimmten Zeitraum so gesetzt. Das Ventil reagiert aber scheinbar überhaupt nicht darauf. 
Das sieht man dann an der Temperaturkurve, die nicht steigt.
Setze ich die Temperatur hoch und setzt das TRV selbst den Wert hoch, wird es wärmer.
Anscheinend kann man die Werte doch gar nicht manuell setzen, z.B. um mit optimaler Heizkurve und möglichst offenen Ventilen zu arbeiten. 
Oder hat das schon jemand in Deiner Erinnerung hinbekommen bzw. gibt es einen Trick, wie man den Öffnungsgrad über die Input-Variable nutzen kann?
Es wäre toll, wenn ich die Input-Entität aktiv nutzen könnte.

Gruß und danke für einen abschließenden Gedanken/Tipp

Sebastian

**Comment by SukramJ** [comment]
> Allerdings habe ich den use case für diese input Variablen nicht verstanden.

1. Wir definieren hier keine use cases. Diese Entität wurde automatisch generiert.
2. Das sind Number Entitäten und keine Input Variablen.

Bei meinem HmIP-eTRV-2 funktioniert die Änderung der Ventilöffnung über LEVEL. Nutze ich nicht, und finde ich auch total überflüssig, aber für einige Benutzer ist das anscheinend wichtig.

**Comment by Bascht74** [comment]
Alles klar, dann frag ich dazu mal in einem passenden Forum nach oder suche nochmal intensiver über Google, wie man das nutzt. Ich würde das auch gerne nutzen :-)

**Comment by Baxxy13** [comment]
Das googeln kannst du dir sparen. 
Ja, man kann den LEVEL setzen und das eTRV fährt dann auch kurz die Position an. Die interne Regelungslogik steuert aber kurz darauf wieder gegen um auf den vorigen LEVEL zurückzukehren. Überlisten, aushebeln oder deaktivieren lässt sich die interne Regelungslogik nicht. 

Den Wunsch, den LEVEL als Sensor darzustellen (wie bei den HM Heizkörperthermostaten), hatte ich indirekt mal in #633 geäußert. Jetzt wo die Number-Entitäten keine Langzeitstatistik mehr haben wird der Wunsch sogar noch wünschenswerter.

> Nutze ich nicht, und finde ich auch total überflüssig, aber für einige Benutzer ist das anscheinend wichtig.

Na dann gib Dir nen Ruck und weg damit. ;-)
Ich kann mir beim besten Willen nicht vorstellen das das jemand nutzt.

**Comment by SukramJ** [comment]
Also ich soll:
1. .. was programmieren das die automatisch generierte `number entity` **stattdessen** als `sensor entity `erstellt werden.
2. ... mich mit den _"Heizungsoptimierern"_ rumschlagen, die die _"volle Kontrolle"_ über Ihr System haben wollen, und auf keinen Fall auf ein editierbares `LEVEL` verzichten können.

Punkt 1 geht ja noch, aber auf Punkt 2 hab ich absolut keine Lust. Ausser jemand ergänzt die Doku mit einem Abschnitt, den wir noch nicht haben, warum der eigentlich schreibbare Wert LEVEL bei einem TRV doch nur als Sensor dargestellt wird.

**Reply by SukramJ** [reply]
@Baxxy13 Könntest Du einen kurzen Text beisteuern, warum die Verwendung von LEVEL (editierbar) bei TRVs nicht sinnvoll ist?

Kann hier in der Diskussion sein.

**Reply by Baxxy13** [reply]
Ja klar, wird aber erst bisschen später was.

Ich hatte ja gehofft jemand bringt mal einen Anwendungsfall bei dem die kurzfristige manuelle Steuerung des `LEVEL` Sinn ergibt.
Mir fällt da nur eine "händische Entkalkungsfahrt" zu ein.

**Comment by Baxxy13** [comment]
Entwurf:

Ventilöffnungsgrad (`LEVEL`) bei HmIP-Heizkörperthermostaten (HmIP-eTRVxxx):
(Integration als Sensor-Entität anstatt Number-Entität)

Der `LEVEL`-Datenpunkt der eTRV's ist beschreibbar, d.h. man kann das Ventil auf eine beliebige Position fahren.
Diese "manuelle Position" wird aber kurz darauf durch die interne Regelungslogik des Gerätes wieder rückgängig gemacht, das Ventil fährt somit fast sofort wieder in die Ursprungsposition. Da sich die interne Regelungslogik des Gerätes weder umgehen noch deaktivieren lässt ist die manuelle Steuerung des Ventilöffnungsgrades nicht sinnvoll nutzbar.

**Reply by SukramJ** [reply]
Danke @Baxxy13, 
das Feature wurde mit [1.21.1](https://github.com/danielperna84/custom_homematic/releases/tag/1.21.1) released, und die Dokumentation entsprechend ergänzt.
